### ISSUE TITLE
deepseekv2转换误差

### ISSUE THREAD

**Issue by liux750** [alibaba/Pai-Megatron-Patch#469]
使用教程中的deepseekv2 lite hf转megatron，误差如下：
~~~
layer:26, layer_in, diff: 20480, diff>1e-05:[10196/20480] diff_max:57.86997985839844
layer:26, q_proj_in, diff: 20480, diff>1e-05:[10286/20480] diff_max:8.381954193115234
layer:26, q_proj_out, diff: 30720, diff>1e-05:[15446/30720] diff_max:2.5908823013305664
layer:26, q_proj_out_weight, diff: 0, diff>1e-05:[0/6291456] diff_max:0.0
layer:26, kv_a_proj_in, diff: 20480, diff>1e-05:[10286/20480] diff_max:8.381954193115234
layer:26, kv_a_proj_out, diff: 5760, diff>1e-05:[2983/5760] diff_max:12.524017333984375
layer:26, kv_a_proj_out_weight, diff: 0, diff>1e-05:[0/1179648] diff_max:0.0
layer:26, kv_a_norm_in, diff: 5120, diff>1e-05:[2652/5120] diff_max:12.524017333984375
layer:26, kv_a_norm_out, diff: 5120, diff>1e-05:[2624/5120] diff_max:3.7540347576141357
layer:26, kv_a_norm_out_weight, diff: 0, diff>1e-05:[0/512] diff_max:0.0
layer:26, kv_b_proj_in, diff: 5120, diff>1e-05:[2624/5120] diff_max:3.7540347576141357
layer:26, kv_b_proj_out, diff: 40960, diff>1e-05:[20437/40960] diff_max:3.1056199073791504
layer:26, kv_b_proj_out_weight, diff: 0, diff>1e-05:[0/2097152] diff_max:0.0
layer:26, o_proj_in, diff: 20480, diff>1e-05:[10081/20480] diff_max:1.679904580116272
layer:26, o_proj_out, diff: 20480, diff>1e-05:[9964/20480] diff_max:3.2377185821533203
layer:26, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:26, attn_out, diff: 20480, diff>1e-05:[9964/20480] diff_max:3.2377185821533203
layer:26, shared_experts_down_proj_in, diff: 28160, diff>1e-05:[14102/28160] diff_max:42.54461669921875
layer:26, shared_experts_down_proj_out, diff: 20480, diff>1e-05:[9978/20480] diff_max:113.819580078125
layer:26, shared_experts_down_proj_out_weight, diff: 0, diff>1e-05:[0/5767168] diff_max:0.0
layer:26, lmhead, diff: 1024000, diff>1e-05:[754661/1024000] diff_max:12.326448440551758
layer:26, lmhead_weight, diff: 0, diff>1e-05:[0/209715200] diff_max:0.0
layer:26, lmhead_token, diff: 5, diff>1e-05:[4/10] diff_max:26968
logits: 1024000, diff>1e-05:[754661/1024000] diff_max:12.326448440551758
~~~

其中layer:26, shared_experts_down_proj_out, diff: 20480, diff>1e-05:[9978/20480] diff_max:113.819580078125  竟然达到了113，确定没问题吗？

**Comment by dubanx** [comment]
你好，请问你转换的误差有影响吗？我转换了一个qwen2.5 3B，误差如下（第一次进行转换，不是很了解转换误差的大体在什么范围）
layer:0, layer_in, diff: 0, diff>1e-05:[0/20480] diff_max:0.0
layer:0, o_proj_in, diff: 13273, diff>1e-05:[6585/20480] diff_max:0.03125
layer:0, o_proj_out, diff: 15475, diff>1e-05:[7696/20480] diff_max:0.0234375
layer:0, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:0, attn_out, diff: 15475, diff>1e-05:[7696/20480] diff_max:0.0234375
layer:1, layer_in, diff: 15467, diff>1e-05:[7809/20480] diff_max:0.0625
layer:1, o_proj_in, diff: 16911, diff>1e-05:[8994/20480] diff_max:0.0234375
layer:1, o_proj_out, diff: 16088, diff>1e-05:[8175/20480] diff_max:0.03125
layer:1, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:1, attn_out, diff: 16088, diff>1e-05:[8175/20480] diff_max:0.03125
layer:2, layer_in, diff: 16555, diff>1e-05:[8338/20480] diff_max:0.1875
layer:2, o_proj_in, diff: 18076, diff>1e-05:[8949/20480] diff_max:0.0478515625
layer:2, o_proj_out, diff: 18638, diff>1e-05:[9446/20480] diff_max:0.07421875
layer:2, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:2, attn_out, diff: 18638, diff>1e-05:[9446/20480] diff_max:0.07421875
layer:3, layer_in, diff: 17344, diff>1e-05:[8887/20480] diff_max:20.0
layer:3, o_proj_in, diff: 19687, diff>1e-05:[10441/20480] diff_max:0.0703125
layer:3, o_proj_out, diff: 19486, diff>1e-05:[9482/20480] diff_max:0.01611328125
layer:3, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:3, attn_out, diff: 19486, diff>1e-05:[9482/20480] diff_max:0.01611328125
layer:4, layer_in, diff: 17640, diff>1e-05:[8929/20480] diff_max:20.0
layer:4, o_proj_in, diff: 18931, diff>1e-05:[9400/20480] diff_max:0.03515625
layer:4, o_proj_out, diff: 18128, diff>1e-05:[9086/20480] diff_max:0.015625
layer:4, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:4, attn_out, diff: 18128, diff>1e-05:[9086/20480] diff_max:0.015625
layer:5, layer_in, diff: 17585, diff>1e-05:[8879/20480] diff_max:20.0
layer:5, o_proj_in, diff: 19393, diff>1e-05:[8624/20480] diff_max:0.03515625
layer:5, o_proj_out, diff: 18580, diff>1e-05:[9362/20480] diff_max:0.01953125
layer:5, o_proj_out_weight, diff: 0, diff>1e-05:[0/4194304] diff_max:0.0
layer:5, attn_out, diff: 18580, diff>1e-05:[9362/20480] diff_max:0.01953125
layer:6, layer_in, diff: 17611, diff>1e-05:[8945/20480] diff_max:16.0
